x-pack/docs/en/security/authentication/saml-guide.asciidoc

```diff
@@ -356,6 +356,34 @@ address such as `admin@staff.example.com.attacker.net`. It is important that
 you make sure your regular expressions are as precise as possible so that
 you do not inadvertently open an avenue for user impersonation attacks.
 
+==== Requesting specific authentication methods
+
+It is sometimes necessary for a SAML SP to be able to impose specific
+restrictions regarding the authentication that will take place at an IdP,
+in order to assess the level of confidence that it can place in
+the corresponding authentication response. The restrictions might have to do
+with the authentication method (password, client certificates, etc), the
+user identification method during registration and others. {es} implements
+https://docs.oasis-open.org/security/saml/v2.0/saml-authn-context-2.0-os.pdf[SAML 2.0 Authentication Context]
+that can be used for this purpose as defined in SAML 2.0 Core Specification.
+
+In short, the SAML SP defines a set of Authentication Context Class Reference
+values describing the restrictions to be imposed on the IdP and sends these
+in the Authentication Request. The IdP attempts to grant these restrictions,
+and either returns an indication of which it satisfied in the Authentication
+Statement of the SAML Response if it can, or it fails to authenticate the
+user if it can't adhere to the restrictions.
```

Comment:
I think this will read better if it's more of a "if-scenario-A, then action-B ..." rather than "_Either_ action-B if scenario-A, or ..."
For example
```
attempts to grant these restrictions,
and if it cannot adhere to the restrictions, then it fails the authentication attempt.
If the user is successfully authenticated, then the Authentication Statement of the
SAML Response will return an indication of the restrictions that were satisfied.
```